extensions/objects/templates/list-of-sets-of-translatable-html-content-ids-editor.component.ts

```diff
@@ -142,7 +142,31 @@ export class ListOfSetsOfTranslatableHtmlContentIdsEditorComponent {
 
   ngOnInit(): void {
     this.initValues = [];
-    this.choices = this.initArgs.choices;
+    try {
+      this.choices = this.initArgs.choices;
+    } catch (err) {
+      // NOTE: This try-catch block is added to
+      // track the issue #14708. However, the error does not
+      // seem to occur anymore. This block adds some
+      // additional details to the error message to debug the error
+      // if it occurs again.
+      const additionalDetails = (
+        '\nAdditional debug info:\n' +
+        'This object editor is related to ' +
+        'editing \'Drag And Drop Sort\' interaction responses. ' +
+        'This component is called and recieves data from ' +
+        'object-editor.directive.ts which in turn is called and ' +
+        'recieves data from rule-editor.component.ts.\n' +
+        'Possible files to look for while debugging: \n' +
+        '1. object-editor.directive.ts\n' +
+        '2. rule-editor.component.ts\n' +
+        '3. responses.service.ts\n' +
+        'Also, this error does not occur frequently and ' +
+        'mostly occurs on production. Hence, use --prod ' +
+        'flag while debugging.'
```

Comment:
I'm a bit confused about this. How will these additional logs help in debugging further when this error occurs again on prod in the future -- it seems to be a hardcoded string?

In general, when adding additional details to logs, think about what variables you want to capture (based on your hypothesis of when this error might arise) so we can use the info logged in the future to confirm/reject the hypothesis and work on a fix.

Reply:
@kevintab95  Got it. I have tried to improve the log now and logged a variable which I think might be responsible for this error (if it occurs again). PTAL.